### ISSUE TITLE
XONTRIBS_AUTOLOAD_DISABLED is not respected

### ISSUE THREAD

**Issue by eugenesvk** [xonsh/xonsh#5020]
I've noticed that xontribs get autoloaded even if

```
❯ cat /Users/Shared/rc.xsh
$XONTRIBS_AUTOLOAD_DISABLED = True
```
## xonfig

<details>

```
$ xonfig
+------------------+----------------------+
| xonsh            | 0.13.4               |
| Python           | 3.10.9               |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 3.0.36               |
| shell type       | prompt_toolkit       |
| history backend  | json                 |
| pygments         | 2.13.0               |
| on posix         | True                 |
| on linux         | False                |
| on darwin        | True                 |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
| xontrib 1        | homebrew             |
| xontrib 2        | cd                   |
| RC file 1        | /Users/Shared/rc.xsh |
+------------------+----------------------+
```

</details>

## Expected Behavior

No xontribs should autoload if `$XONTRIBS_AUTOLOAD_DISABLED` is set

## Current Behavior
The `$XONTRIBS_AUTOLOAD_DISABLED` is not respected even if it's the first (and only) thing in user's `rc.xsh`

### Traceback (if applicable)

<details>

```
traceback
```

</details>

## Steps to Reproduce
Add to an empty `rc.xsh` file `$XONTRIBS_AUTOLOAD_DISABLED=True` and launch xonsh pointing to this file

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by eugenesvk** [comment]
I think this is due to the fact that the autoload phase happens too early, so another issue is that the env vars aren't set up properly from the user files, so if your xontrib requires user input, it's out of luck and shouldn't be autoloaded, but instead ask the user to load itself manually

**Comment by jnoortheen** [comment]
@eugenesvk could you try setting it before starting the xonsh shell? I must have tested that way. 
`env XONTRIBS_AUTOLOAD_DISABLED=1 xonsh`

**Comment by eugenesvk** [comment]
I unload the autoloaded xontrib, execute your command, and the xontrib gets autoloaded (the env var is set to 1)
(though this is also an impractical way to set an env var)

**Comment by dysfungi** [comment]
I think this issue is good to close as I created a dupe, https://github.com/xonsh/xonsh/issues/5872, that got addressed in https://github.com/xonsh/xonsh/pull/5873

**Comment by jnoortheen** [comment]
Good find @dysfungi !
